src/main/java/Command.java

```diff
@@ -0,0 +1,16 @@
+public abstract class Command {
+    protected String line;
+    int itemNo = -1;
```

Comment:
Perhaps it would be better to just let the child classes that require this variable implement it instead? If not it would be slightly redundant for the other classes to contain it.